src/utils/session.ts

```diff
@@ -1,24 +1,46 @@
+import { defaultDetails } from "../constants";
 import { Request } from "express";
-import { isDetails } from "private-api-sdk-node/dist/services/presenter-account/types";
-import { type Details } from "private-api-sdk-node/src/services/presenter-account/types";
+import { Name, type Details } from "private-api-sdk-node/src/services/presenter-account/types";
 
 export const PRESENTER_ACCOUNT_SESSION_KEY = "presenter_account_details";
 
 export function getPresenterAccountDetails(req: Request): Details {
-    const presenterAccountDetails = req.session?.getExtraData(PRESENTER_ACCOUNT_SESSION_KEY);
-
-    if (presenterAccountDetails === undefined) {
-        throw new Error(`Presenter account details are undefined`);
+    let presenterAccountDetails: Details | undefined;
+    if (req.session !== undefined){
+        presenterAccountDetails = req.session.getExtraData(PRESENTER_ACCOUNT_SESSION_KEY);
+        if (presenterAccountDetails === undefined) {
+            presenterAccountDetails =  fetchUserDetails(req, defaultDetails);
+            setPresenterAccountDetails(req, defaultDetails);
+        }
+        return presenterAccountDetails;
     }
+    return fetchUserDetails(req, defaultDetails);
```

Comment:
These changes remove the type checking.
It also returns the default details if they are not present this is not what I'd expect this function to do. It should just get the presenter account details. 
If the type checking fails it should throw an error, or return undefined if the details are not repsenter.
The use of if statements is overly verbose and distracts from the logic. It should use nullish coalescing to make it clearer.